### PR TITLE
feat(p2p): add wire package for messages io

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

File: src/network/wire/lib.zig [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+/// Use it on serialized messages to compute the header's value
+fn computePayloadChecksum(payload: []u8) [4]u8 {
+    var digest: [32]u8 = undefined;
+    Sha256.hash(payload, &digest, .{});
+    Sha256.hash(&digest, &digest, .{});
+
+    var ret: [4]u8 = undefined;
+    @memcpy(ret[0..], digest[0..4]);
+
+    return ret;
```
Comment (StringNick) [inline src/network/wire/lib.zig:30]: u can just  use 
```return digest[0..4].*;```

File: src/network/protocol/lib.zig [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 pub const messages = @import("./messages/lib.zig");
 
-/// Protocol version
-pub const PROTOCOL_VERSION: i32 = 70015;
-
 /// Known network ids
-pub const NetworkMagicBytes = struct {
-    pub const MAINNET: [4]u8 = 0xd9b4bef9;
+pub const BitcoinNetworkId = struct {
+    pub const MAINNET: [4]u8 = [4]u8{ 0xd9, 0xb4, 0xbe, 0xf9 };
```
Comment (StringNick) [inline src/network/protocol/lib.zig:5]:
```suggestion
    pub const MAINNET: [4]u8 = .{ 0xd9, 0xb4, 0xbe, 0xf9 };
```

File: src/network/protocol/messages/version.zig [hunk 3]
```diff
@@ -2,200 +2,224 @@ const std = @import("std");
...
+    version: i32,
     start_height: i32,
+    recv_port: u16,
+    trans_port: u16,
+    user_agent: ?[]const u8,
     relay: ?bool,
 
-    /// Will free the user_agent if present in the message
+    pub fn name() [12]u8 {
+        var ret = std.mem.zeroes([12]u8);
```
Comment (StringNick) [inline src/network/protocol/messages/version.zig:31]:
```suggestion
pub inline fn name() [12]u8 {
     return protocol.CommandNames.VERSION ++ [_]u8{0} ** 5;
}
```

File: src/network/protocol/messages/version.zig [hunk 4]
```diff
@@ -2,200 +2,224 @@ const std = @import("std");
...
+    ///
+    /// Computed as `Sha256(Sha256(self.serialize()))[0..4]`
+    pub fn checksum(self: VersionMessage) [4]u8 {
+        var digest: [32]u8 = undefined;
+        var hasher = Sha256.init(.{});
+        const writer = hasher.writer();
+        self.serializeToWriter(writer) catch unreachable; // Sha256.write is infaible
+        hasher.final(&digest);
+
+        Sha256.hash(&digest, &digest, .{});
```
Comment (StringNick) [inline src/network/protocol/messages/version.zig:46]: i am not sure, but in bitcoin-primitives should double sha256

File: src/network/protocol/messages/version.zig [hunk 5]
```diff
@@ -2,200 +2,224 @@ const std = @import("std");
...
+                return false;
+            }
         }
-    }
 
-    // relay
-    if (lhs.relay) |ln| {
-        if (rhs.relay) |rn| {
-            if (ln != rn) {
+        // relay
```
Comment (StringNick) [inline src/network/protocol/messages/version.zig:203]: i think for this part u can use std.meta.eql

File: src/network/wire/lib.zig (hunk 1 above)
Comment (tdelabro) [reply src/network/wire/lib.zig:30]: done

File: src/network/protocol/lib.zig (hunk 2 above)
Comment (tdelabro) [reply src/network/protocol/lib.zig:5]: done

File: src/network/protocol/messages/version.zig (hunk 4 above)
Comment (tdelabro) [reply src/network/protocol/messages/version.zig:46]: I looked for an exposed method in bitcoin-primitives doing that but I couldn't find it

**Pull request merged**